test/blocks/global-footer/global-footer.test.js

```diff
@@ -0,0 +1,247 @@
+import placeholders from './mocks/placeholders.js';
+
+const originalFetch = window.fetch;
+
+describe('global footer', () => {
+  let clock = null;
+  beforeEach(async () => {
+    document.body.innerHTML = baseFooter;
+    clock = sinon.useFakeTimers({
+      toFake: ['setTimeout'],
+      shouldAdvanceTime: true,
+    });
+
+    window.fetch = stub().callsFake((url) => {
+      if (url.includes('/footer')) {
+        return mockRes({
+          payload: fetchedFooter(
+            { regionPickerHash: '/fragments/regions#langnav' },
+          ),
+        });
+      }
+
+      if (url.includes('/placeholders')) return mockRes({ payload: placeholders });
+
+      if (url.includes('icons.svg')) return mockRes({ payload: icons });
+
+      return null;
+    });
+  });
+
+  afterEach(() => {
+    clock.restore();
+    window.fetch = originalFetch;
+  });
+
+  describe('wide screen', async () => {
+    it('should render the footer on wide screen', async () => {
+      await createFullGlobalFooter({ waitForDecoration: true, viewport: 'wide' });
+
+      Object.keys(allSelectors).forEach((key) => expect(
+        document.querySelector(allSelectors[key]) instanceof HTMLElement,
+      ).to.equal(true));
+
+      expect(allElementsVisible(
+        visibleSelectorsDesktop,
+        document.querySelector(containerSelector),
+      )).to.equal(true);
+    });
+  });
+
+  describe('desktop', () => {
+    describe('basic sanity tests', () => {
+      it('should have footer', async () => {
+        await createFullGlobalFooter({ waitForDecoration: true });
+        expect(document.querySelector('footer')).to.exist;
+
+        Object.keys(allSelectors).forEach((key) => expect(
+          document.querySelector(allSelectors[key]) instanceof HTMLElement,
+        ).to.equal(true));
+
+        expect(allElementsVisible(
+          visibleSelectorsDesktop,
+          document.querySelector(containerSelector),
+        )).to.equal(true);
+      });
+    });
+
+    describe('conditional render tests', () => {
+      it('should render the footer when in viewport', async () => {
+        await createFullGlobalFooter({ waitForDecoration: true });
+
+        Object.keys(allSelectors).forEach((key) => expect(
+          document.querySelector(allSelectors[key]) instanceof HTMLElement,
+        ).to.equal(true));
+
+        expect(allElementsVisible(
+          visibleSelectorsDesktop,
+          document.querySelector(containerSelector),
+        )).to.equal(true);
+      });
+
+      it('should render the footer after 3s when outside of the 300px range of the viewport', async () => {
+        insertDummyElementOnTop({ height: window.innerHeight + 400 });
+
+        await createFullGlobalFooter({ waitForDecoration: false });
+
+        Object.keys(allSelectors).forEach((key) => expect(
+          document.querySelector(allSelectors[key]) instanceof HTMLElement,
+        ).to.equal(false));
+
+        clock.tick(3000);
```

Comment:
Do we need to be this specific or would `await clock.runAllAsync();` also do the trick?

Reply:
In here I think we need to be specific because the test is checking if the footer is waiting for 3s before rendering, when outside of the 300px range of the viewport. This 3s seconds wait is a feature of the footer.